Sync/Synchronizers/Bookmarks/BookmarksRepairRequestor.swift

```diff
@@ -0,0 +1,535 @@
+        self.scratchpad = scratchpad
+        self.prefs = basePrefs.branch("repairs.bookmark")
+        self.remoteClients = remoteClients
+    }
+
+    /**
+     * See if the repairer is willing and able to begin a repair process given
+     * the specified validation information.
+     *
+     * - returns: true if a repair was started and false otherwise.
+     */
+    func startRepairs(validationInfo: [(type: String, ids: [String])], flowID: String = Bytes.generateGUID()) -> Deferred<Maybe<Bool>> {
+        if (self.currentState != RepairState.notRepairing) {
+            log.info("Can't start a repair - repair with ID \(self.flowID) is already in progress")
+            return deferMaybe(false)
+        }
+
+        let ids = self.getProblemIDs(validationInfo)
+
+        if (ids.count == 0) {
+            log.info("Not starting a repair as there are no problems")
+            return deferMaybe(false)
+        }
+
+        if (ids.count > MaxRequestedIDs) {
+            log.info("Not starting a repair as there are over \(MaxRequestedIDs) problems")
+            // TODO: TELEMETRY
+            // let extra = { flowID, reason: `too many problems: ${ids.size}` }
+            // this.service.recordTelemetryEvent("repair", "aborted", undefined, extra)
+            return deferMaybe(false)
+        }
+
+        return self.anyClientsRepairing() >>== { clientsRepairing in
+            if (clientsRepairing) {
+                log.info("Can't start repair, since other clients are already repairing bookmarks")
+                // TODO: TELEMETRY
+                // let extra = { flowID, reason: "other clients repairing" }
+                // this.service.recordTelemetryEvent("repair", "aborted", undefined, extra)
+                return deferMaybe(false)
+            }
+
+            log.info("Starting a repair, looking for \(ids.count) missing item(s)")
+            // Setup our prefs to indicate we are on our way.
+            self.flowID = flowID
+            self.currentIDs = Array(ids)
+            self.currentState = RepairState.needNewClient
+            // TODO: TELEMETRY
+            // this.service.recordTelemetryEvent("repair", "started", undefined, { flowID, numIDs: ids.size.toString() })
+            return self.continueRepairs()
+        }
+    }
+
+    /**
+     * Work out what state our current repair request is in, and whether it can
+     * proceed to a new state.
+     *
+     * - returns: true if we could continue the repair - even if the state didn't
+     *            actually move. Returns false if we aren't actually repairing.
+     */
+    func continueRepairs(response: RepairResponse? = nil) -> Deferred<Maybe<Bool>> {
+        // Note that "aborted" and "finished" should never be current when this
+        // function returns - this function resets to notRepairing in those cases.
+        if (self.currentState == RepairState.notRepairing) {
```

Comment:
Nit: Swift is smart enough to infer the enum type here so you can just use `.notRepairing`